backend/src/common/client/api_server/v2/experiment_client.go

```diff
@@ -0,0 +1,217 @@
+
+	// Creating experiment client
+	return &ExperimentClient{
+		apiClient:      apiClient,
+		authInfoWriter: api_server.SATokenVolumeProjectionAuth,
+	}, nil
+}
+
+func (c *ExperimentClient) Create(parameters *params.CreateExperimentParams) (*model.V2beta1Experiment,
+	error) {
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), apiServerDefaultTimeout)
+	defer cancel()
+
+	// Make service call
+	parameters.Context = ctx
+	response, err := c.apiClient.ExperimentService.CreateExperiment(parameters)
+	if err != nil {
+		return nil, util.NewUserError(err,
+			fmt.Sprintf("Failed to create experiment. Params: '%+v'. Body: '%+v'", parameters, parameters.Body),
+			fmt.Sprintf("Failed to create experiment '%v'", parameters.Body.DisplayName))
+	}
+
+	return response.Payload, nil
+}
+
+func (c *ExperimentClient) Get(parameters *params.GetExperimentParams) (*model.V2beta1Experiment,
+	error) {
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), apiServerDefaultTimeout)
+	defer cancel()
+
+	// Make service call
+	parameters.Context = ctx
+	response, err := c.apiClient.ExperimentService.GetExperiment(parameters)
+	if err != nil {
+		return nil, util.NewUserError(err,
+			fmt.Sprintf("Failed to get experiment. Params: '%+v'", parameters),
+			fmt.Sprintf("Failed to get experiment '%v'", parameters.ExperimentID))
+	}
+
+	return response.Payload, nil
+}
+
+func (c *ExperimentClient) List(parameters *params.ListExperimentsParams) (
+	[]*model.V2beta1Experiment, int, string, error) {
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), apiServerDefaultTimeout)
+	defer cancel()
+
+	// Make service call
+	parameters.Context = ctx
+	response, err := c.apiClient.ExperimentService.ListExperiments(parameters)
+	if err != nil {
+		return nil, 0, "", util.NewUserError(err,
+			fmt.Sprintf("Failed to list experiments. Params: '%+v'", parameters),
+			fmt.Sprintf("Failed to list experiments"))
+	}
+
+	return response.Payload.Experiments, int(response.Payload.TotalSize), response.Payload.NextPageToken, nil
+}
+
+func (c *ExperimentClient) Delete(parameters *params.DeleteExperimentParams) error {
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(context.Background(), apiServerDefaultTimeout)
+	defer cancel()
+
+	// Make service call
+	parameters.Context = ctx
+	_, err := c.apiClient.ExperimentService.DeleteExperiment(parameters)
+	if err != nil {
+		return util.NewUserError(err,
+			fmt.Sprintf("Failed to delete experiments. Params: '%+v'", parameters),
+			fmt.Sprintf("Failed to delete experiment"))
+	}
+
+	return nil
+}
+
+func (c *ExperimentClient) ListAll(parameters *params.ListExperimentsParams, maxResultSize int) (
+	[]*model.V2beta1Experiment, error) {
+	return listAllForExperiment(c, parameters, maxResultSize)
+}
+
+func listAllForExperiment(client ExperimentInterface, parameters *params.ListExperimentsParams,
```

Comment:
For my education, why do we need `listAllForExperiment` function and the ` to to wrap this function?